### DISCUSSION TITLE
TRV11 v2.0.1 - Patch version

### DISCUSSION THREAD

**Discussion by Rishabh-ondc** [ONDC-Official/mobility-specification#195] [General]
Features addressed as part of the v2.0.1 patch release are mentioned below.

- Timestamp at item level signifying validity of ticket - Example available in draft-TRV11-2.0.1
  - item.time.duration or item.time.timestamp

- Fulfilment objects - TRIP and TICKET - Example available in draft-TRV11-2.0.1
  - Trip signifying journey 
  - Ticket signifying individual tickets (viz. single or multiple qr codes)
  - parent_id to be leveraged for hierarchy. A trip would have ticket(s)

- Vehicle based confirmation flow
  - update API flow with vehicle number for vehicle number based confirmation flow. Example available in draft-TRV11-2.0.1

- Pagination logic for large catalogs
  - pagination logic provided as examples for sharing large catalogs as part of on_search payloads. Example available in draft-TRV11-2.0.1

- Unlimitted passes
  - example added to illustrate transaction for buying passes for unlimitted passes

**Comment by AmanChartr** [comment]
Hi, @Rishabh-ondc this is regarding on_search Payload Pagination.

When dealing with the `on_search` payload, we want to optimise the structure by reducing redundancy, particularly with the stops object. Here's the approach:

- **Redundancy in Stops:** The same stop may appear in multiple routes, causing unnecessary repetition of the stops object in each route.

- **Solution - Separate Stops Array:** To eliminate this redundancy, we can create a separate array that contains all the unique stops (stops objects) that are used across the routes on the current page.

- **Unique Stop ID:** Each stop object in this array will be assigned a unique identifier (ID) that acts as a reference for that stop.

- **Mapping Stops to Routes:** Instead of embedding the complete stops object in each route within the fulfillments array, the routes will only reference the corresponding stop using its unique ID from the stops array.

This way, we reduce redundancy and optimise the data structure by storing each stop object only once, while maintaining the ability to reference it efficiently for each route and will also reduce the size of the json.

Let me know if this makes sense or if you need further clarification!
